Add "About you" and "Your family" pages to the user flow

pcraig3 opened the pull request:
This PR adds two new pages to the user flow: an "About you" page and a "Your family" page. 
On these pages, users will have the opportunity to change their (a) name and address or (b) marital status or number of kids. (Note that the "change" links don't actually work yet.)

It fulfills all the tasks in the ["Online confirmation: Personal information verification" card on Trello](https://trello.com/c/gmrgDMUg/35-online-confirmation-personal-information-verification) (except for the functional tests, and there's another PR up for that).

# Screenshots

## "About you" page 

![localhost_3000_about-you](https://user-images.githubusercontent.com/2454380/56315087-d7dfdc00-6124-11e9-87b4-c8a28f43531e.png)

## "Your family" page

![localhost_3000_your-family](https://user-images.githubusercontent.com/2454380/56614867-fee85300-65e7-11e9-8da8-54e016bab46d.png)

pcraig3 committed:
Remove Form submission configuration from the Dashboard page

Removed the 'POST' route definition in server.js and removed the
ErrorList object from the Dashboard as well.

The Dashboard no longer contains the consent form, so we don't need
any of this anymore. Makes the page simpler.

pcraig3 committed:
First stab at an AboutYou page

Still have to generalize the CSS and add some basic tests, but it
mostly works.

pcraig3 committed:
Move dashboard styles out into styles.js

There's maybe a better name for this, but for now this is fine.

pcraig3 committed:
Add basic unit tests for AboutYou page

Test rendering our current "About You" page.

Make sure that it has the heading we expect, a "continue" button,
a link to log out, and that if we pass data into it that we
can see that data on the page.

pcraig3 commented:
Update: I've shown this to @caycefischer and Nancy and both have given the green light.

sboisvert commented on src/pages/AboutYou.js:
```diff
@@ -0,0 +1,36 @@
+        <${SummaryTable} rows=${aboutYouRows(data)} ifEditable=${true} //>
+        <p>
+          There are <strong>2 sections</strong> remaining, which should take ${' '}<strong
+            >5 minutes</strong
```
I'm not sure I fully understand the spacing on this line and the next 2. Is it a pattern I'm not aware of?

pcraig3 replied:
Oh, that's my editor formatting the line.

Once I put `<strong>` tags around some of the words, it formatted it like:

```
<p>
  There are <strong>2 sections</strong> remaining, which should take
  <strong>5 minutes</strong>
  to complete.
</p>
```

And then on the screen it looked like: "There are **2 sections** remaining, which should take**5 minutes**to complete".

So I added forced whitespace: `${' '}` around the `strong` tags and that's where we ended up with.

sboisvert commented on src/pages/AboutYou.js:
```diff
@@ -0,0 +1,36 @@
+        <${LogoutLink} />
+        <h1>About you</h1>
+        <p>
+          This is the current name and address we have on file for you. Please update any
```
Are we putting the marital and dependents information on a different screen?

pcraig3 replied:
Yes, it's on a different screen** (Sorry, I was confused.)

![image](https://user-images.githubusercontent.com/2454380/56605306-d5bcc800-65d1-11e9-9ab8-8a1b572a0654.png)

pcraig3 committed:
Add employer name to dashboard page

Since we now have dynamic employer names, we shouldn't hardcode it
anymore.

I've made them all uppercase because it's more fun.

pcraig3 committed:
Bring dashboardStyles into T4 page

This reduces some copy+paste CSS we had.

pcraig3 committed:
Refactor Dashboard test

We basically want a flattened array from the data object.

pcraig3 committed:
Add a page for family information

Looks a lot like the About You page. I'm sure we can optimize this
if we need to but I think this will probably change in future.

Also, moved the "T4" route down in the server file to more
or less resemble the flow through the app.

pcraig3 committed:
Add tests for your family page

Looks a lot like our About Page tests.
Check that the H1 exists and that the data that we pass to the page
component shows up on the page somewhere.

pcraig3 committed:
Use sentence case for income page

I didn't notice before, but all of our other page headings are in
sentence case, so this page should be as well.

Also the GDS design system says "Write all headings in sentence case."

Reference:
- https://design-system.service.gov.uk/styles/typography/#headings

pcraig3 committed:
Write what the boxes represent instead of writing "box 12"

According to Nancy:
- Box 12 is SIN
- Box 14 is Employment Income
- Box 22 is Income Tax deducted

pcraig3 committed:
Add 100% width to image.

Not that important but it keeps our page responsive.

pcraig3 committed:
Close <Layout> with <//> so we get our formatting back

For whatever reason, the editor doesn't like it when you close a
custom component with the <${ComponentName}> syntax.
Instead, it prefers <//>, which is alluded to in their README.

- https://github.com/developit/htm#improvements-over-jsx
